packages/modal/src/Modal.tsx

```diff
@@ -0,0 +1,281 @@
+import React, { Fragment, useCallback, useRef, SetStateAction } from 'react';
+import PropTypes from 'prop-types';
+import Portal from '@leafygreen-ui/portal';
+import Icon, { Size } from '@leafygreen-ui/icon';
+import { useEventListener } from '@leafygreen-ui/hooks';
+import { css, cx } from '@leafygreen-ui/emotion';
+
+export const ModalSize = {
+  XXSmall: 'xxsmall',
+  XSmaill: 'xsmall',
+  Small: 'small',
+  Normal: 'normal',
+  Large: 'large',
+  XLarge: 'xlarge',
+} as const;
+
+export type ModalSize = typeof ModalSize[keyof typeof ModalSize];
+
+const overlayStyle = css`
+  animation: fade-in 500ms cubic-bezier(0.165, 0.84, 0.44, 1);
+  -webkit-animation: fade-in 500ms cubic-bezier(0.165, 0.84, 0.44, 1);
+  background-color: #000;
+  opacity: 0.5;
+  position: fixed;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+`;
+
+const modalContentStyle = css`
+  margin: 18px 40px;
+  max-width: 1270px;
+  min-width: 944px;
+  padding: 30px;
+  align-self: flex-start;
+  text-align: left;
+  color: black;
+  position: relative;
+  background-color: #fff;
+  background-clip: padding-box;
+  border: 1px solid rgba(0, 0, 0, 0.1);
+  border-radius: 3px;
+  outline: 0;
+  box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);
+`;
+
+const modalBodyStyle = css`
+  padding: 18px 0;
+`;
+
+const modalSizes: { readonly [K in ModalSize]: string } = {
+  xxsmall: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 366px;
+  `,
+
+  xsmall: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 418px;
+  `,
+
+  small: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 562px;
+  `,
+
+  normal: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 700px;
+  `,
+
+  large: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 800px;
+  `,
+
+  xlarge: css`
+    min-width: inherit;
+    max-width: inherit;
+    width: 1270px;
+  `,
+};
+
+const closeButton = css`
+  font-weight: 200;
+  color: #464c4f;
+  position: relative;
+  float: right;
+  top: -18px;
+  right: -10px;
+  margin-top: -2px;
+  cursor: pointer;
+`;
+
+const titleStyle = css`
+  margin: 0;
+`;
+
+const EscapeKey = 27;
```

Comment:
Let's camelCase this since it's not a class or type. Second, let's move this to where it's used. Lastly, I'd rename to `escapeKeyCode` since that's more accurate.

Reply:
Had it cased like this and in this position per @hswolff's feedback on the Menu Component. Happy to go with whatever you both prefer

Reply:
@hswolff Could you elaborate on why PascalCase is better here?

Reply:
I don't remember making a case for either one, and if I did I disagree with my former self. My main desire was to have it be a named variable for clear referencing.

There is a future I see where we move this to `lib` and have a `const KeyboardKeys = { Escape: 27 }` object for uniform referencing.

Reply:
👍